File: CPP/soal1.cpp
/*
Nama Anddota : 
 - Fasya Nurina Izzati (140810200052)
 - Amariel Danendra Dagna (140810200058)
 - Alya Raisa Hidayat (140810200060)
 - Muthia Azzahra (140810200066)
Kelas           : B
Tanggal Buat    : 1 November 2021
Deskripsi       : Soal 1 - CPP
*/

#include <iostream>

using namespace std;

class Bintang{
    private:
        int baris;

    public:
        Bintang(){
            this->baris = 0;
        }

        Bintang(int baris){
            this->baris = baris;
        }

        void loopFor(){
            int i = 0;
            for(i=0; i<this->baris; i++) {
                cout << (i+1) << ". ";
                for (int j=0; j<this->baris; j++){
                    if (j<i){
                        cout << " ";
                    } else{
                        cout << "*";
                    }
                }
                cout << endl;
            }
        }  

        void loopWhile(){
            int i=0;
            while (i<this->baris){
                cout << (i+1) << ". ";
                int j=0;
                while (j<this->baris){
                    if (j<i){
                        cout << " ";
                    } else {
                        cout << "*";
                    }
                    j++;
                }
                cout << endl;
                i++;
            }
        } 
};

int main(){
    int baris;

    cout << "----------------------------------------" << endl;
    cout << "Masukan Banyak Baris : ";
    cin >> baris;

    Bintang *forBintang = new Bintang(baris);
    Bintang *whileBintang = new Bintang(baris);

    cout << "----------------------------------------" << endl;
    cout << "Bintang dengan Loop For" << endl;
    forBintang->loopFor();

    cout << "----------------------------------------" << endl;
    cout << "Bintang dengan Loop While" << endl;
    whileBintang->loopWhile();
    cout << "----------------------------------------" << endl;
}

File: PYTHON/soal1.py
# Nama Anggota : 
#  - Fasya Nurina Izzati (140810200052)
#  - Amariel Danendra Dagna (140810200058)
#  - Alya Raisa Hidayat (140810200060)
#  - Muthia Azzahra (140810200066)
# Kelas           : B
# Tanggal Buat    : 1 November 2021
# Deskripsi       : Soal 1 - Python

class Bintang:
    def __init__(self, baris):
        self.__baris = baris

    def inputBaris(self):
        self.__baris = int(input("Masukan Banyak Baris : "))

    def loopFor(self):
        for i in range (0, self.__baris):
            print(str(i+1) + ". ", end = '')
            for j in range (0, self.__baris):
                if(j<i):
                    print("  ", end = '')
                else:
                    print('* ', end = '')
            print()

    def loopWhile(self):
        i = 1
        while(i <= self.__baris):
            print(str(i) + ". ", end = '')
            j = 1
            while(j <= self.__baris):
                if(j<i):
                    print("  ", end = '')
                else:
                    print('* ', end = '')
                j += 1
            print()
            i += 1

if __name__ == "__main__":
    star = Bintang(0)
    print("----------------------------------")
    star.inputBaris()
    print("----------------------------------")
    print("Bintang dengan Loop For\n")
    star.loopFor()
    print("----------------------------------")
    print("Bintang dengan Loop While")
    star.loopWhile()


